xdsl/parser/base_parser.py

```diff
@@ -293,12 +293,19 @@ def parse_str_enum(self, enum_type: type[_EnumType]) -> _EnumType:
     def parse_optional_str_enum(self, enum_type: type[_EnumType]) -> _EnumType | None:
         """Parse a string enum value, if present."""
 
-        if self._current_token.kind != MLIRTokenKind.BARE_IDENT:
+        if self._current_token.kind == MLIRTokenKind.BARE_IDENT:
+            val = self._current_token.text
+            if val in enum_type.__members__.values():
+                self._consume_token(MLIRTokenKind.BARE_IDENT)
+                return enum_type(val)
+            return None
+
+        if self._current_token.kind != MLIRTokenKind.STRING_LIT:
             return None
 
-        val = self._current_token.text
+        val = StringLiteral.from_span(self._current_token.span).string_contents
```

Comment:
Feels like we might want a `parse_optional_identifier_or_string_literal` helper for use here and in the attribute dictionary parsing?